cmd/zar/index/command.go

```diff
@@ -30,17 +39,40 @@ func init() {
 
 type Command struct {
 	*root.Command
+	dir   string
+	quiet bool
 }
 
 func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
 	c := &Command{Command: parent.(*root.Command)}
+	f.StringVar(&c.dir, "d", "", "directory to descend")
+	f.BoolVar(&c.quiet, "q", false, "don't print progress on stdout")
 	return c, nil
 }
 
 func (c *Command) Run(args []string) error {
-	if len(args) != 1 {
-		return errors.New("zar index: exactly one directory must be specified")
+	if len(args) == 0 {
+		return errors.New("zar index: one or more indexing patterns must be specified")
+	}
+	if c.dir == "" {
+		return errors.New("zar index: a directory must be specified with -d")
+	}
+	var rules []archive.Rule
+	for _, pattern := range args {
+		rule, err := archive.NewRule(pattern)
+		if err != nil {
+			return errors.New("zar index: " + err.Error())
+		}
+		rules = append(rules, rule)
+	}
+	var progress chan string
+	if !c.quiet {
+		progress = make(chan string)
+		go func() {
+			for line := range progress {
+				fmt.Println(line)
+			}
+		}()
 	}
-	dir := args[0]
-	return archive.CreateIndexes(dir)
+	return archive.IndexDirTree(c.dir, rules, progress)
```

Comment:
Nit: While unlikely, it's possible for archive.IndexDirTree to return (and for the process to exit) before the progress goroutine print the final message. Adding a sync.WaitGroup would fix that.

(This also applies to the hits goroutine in cmd/zar/find/command.go.)

Reply:
goroutines 101

Reply:
Fixed same bug in "zar find" too.